### PR TITLE
Makes this usable as a GitHub Action

### LINKED ISSUE #3: Create GitHub action
It's much easier to use without installing etc.

### PR DESCRIPTION
**Pull request by donatj**
Fixes #3

I am using this tool for projects that aren't written in PHP, and having to set up PHP in my workflow file is … strange. 

This is the basic framework for setting this up as a GitHub Action.

Here is a test pull request showing it in use

https://github.com/donatj/action-test/pull/1/files

The example warnings are just nonsense to test it, the PHP file has nothing to do with the warnings at hand.

image

I am currently _testing_ a specific hash on that PR,

```
donatj/annotate-pull-request-from-checkstyle@<hash>
```

That would become the following once this is merged and tagged and published in the Market

```
staabm/annotate-pull-request-from-checkstyle@v{insert tag}
```

I am more than happy to answer any questions or make any changes you might request.

### CODE REVIEW

**Comment by staabm** [comment]
Thanks for the PR. Would be great to add a example into the Readme

**Comment by donatj** [comment]
Sure thing! I'll give that a quick go.

**Comment by donatj** [comment]
Ok, so I've pushed a little update to the README. I'm not sure how you want to do the tagging, so I left that portion … unfulfilled for now

---

Here's kind of the rub

[https://docs.github.com/en/actions/creating-actions/releasing-and-maintaining-actions#…](https://docs.github.com/en/actions/creating-actions/releasing-and-maintaining-actions#:~:text=we%20recommend%20creating%20releases%20using%20semantically%20versioned%20tags%20%E2%80%93%20for%20example%2C%20v1.1.3%20%E2%80%93%20and%20keeping%20major%20(v1)%20and%20minor%20(v1.1)%20tags%20current%20to%20the%20latest%20appropriate%20commit.%20for%20more%20information%2C%20see%20%22about%20custom%20actions%22%20and%20%22about%20semantic%20versioning.)

> We recommend creating releases using semantically versioned tags – for example, v1.1.3 – and keeping major (v1) and minor (v1.1) tags current to the latest appropriate commit. For more information, see “About custom actions” and “About semantic versioning.

Basically GitHub wants you have a `v1` and for every minor version say `v1.8` tag. They want you to basically delete and replace those tags every time you push a new semantic release.

I'm honestly not 100% certain if that'll cause problems with Packagist/Composer - I've never tried mixing the two before. 

I might recommend breaking action into a separate repo, say a `staabm/annotate-pull-request-from-checkstyle-action` so you can tag/version it separately. It would only take a little rejiggering of the Dockerfile to make that work and I'd be more than happy to help with that.

**Comment by staabm** [comment]
I see.

I have just created https://github.com/staabm/annotate-pull-request-from-checkstyle-action - could you do the necessary plumbing?

**Comment by donatj** [comment]
Yep! I can put something together later today. I'm pretty busy the next couple hours, but I should have a little time tonight.

**Comment by donatj** [comment]
@staabm I've got a little project thrown together. However, GitHub won't let me fork the repo until you've pushed at least one branch. I even tried CLI and it gave me 

```
failed to fork: HTTP 403: The repository exists, but it contains no Git content. Empty repositories cannot be forked. 
(https://api.github.com/repos/staabm/annotate-pull-request-from-checkstyle-action/forks)
```

If you could throw up an empty branch, maybe include just a `LICENSE` that'd be sufficient.

**Comment by staabm** [comment]
just added a License.md

**Comment by donatj** [comment]
This pull request has been updated to be *just* instructions for the README.

The Action code itself has been moved to this PR

https://github.com/staabm/annotate-pull-request-from-checkstyle-action/pull/1

File: README.md [hunk 1]
```diff
@@ -181,6 +181,46 @@ jobs:
...
+on:
+  - pull_request
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: staabm/annotate-pull-request-from-checkstyle@{latest-tag}
```
Comment (staabm) [inline README.md:201]:
```suggestion
      - uses: staabm/annotate-pull-request-from-checkstyle-action@v1
```

File: README.md [hunk 2]
```diff
@@ -181,6 +181,46 @@ jobs:
                 vendor/bin/phpstan analyse --error-format=checkstyle | vendor/bin/cs2pr
 ```
 
+## Using cs2pr as a GitHub Action
+
+You can also use `cs2pr` itself as a GitHub Action. This is useful if you want to for instance use it for a project that does not use PHP or if you want to use it with a custom PHP installation.
```
Comment (staabm) [inline README.md:186]:
```suggestion
You can also use [`cs2pr` itself as a GitHub Action](https://github.com/staabm/annotate-pull-request-from-checkstyle-action). This is useful if you want to for instance use it for a project that does not use PHP or if you want to use it with a custom PHP installation.
```

File: README.md [hunk 3]
```diff
@@ -181,6 +181,46 @@ jobs:
...
+
+### Example
+
+```yml
+name: CS2PR
```
Comment (staabm) [inline README.md:191]:
```suggestion
name: cs2pr
```

File: README.md [hunk 4]
```diff
@@ -181,6 +181,46 @@ jobs:
...
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - uses: staabm/annotate-pull-request-from-checkstyle@{latest-tag}
+        with:
+          files: checkstyle.xml
+          notices-as-warnings: true # optional
+```
```
Comment (staabm) [inline README.md:205]:
```suggestion
You can also use [`cs2pr` itself as a GitHub Action](https://github.com/staabm/annotate-pull-request-from-checkstyle-action). This is useful if you want to for instance use it for a project that does not use PHP or if you want to use it with a custom PHP installation.

See the example at the [cs2pr GitHub Action repositiory](https://github.com/staabm/annotate-pull-request-from-checkstyle-action).

File: README.md [hunk 5]
```diff
@@ -181,6 +181,46 @@ jobs:
...
+- **colorize** _(Optional)_ - Colorize the output (still compatible with Github Annotations)
+  - Expected value: "true"
+- **notices-as-warnings** _(Optional)_ - Convert notices to warnings (Github does not annotate notices otherwise)
+  - Expected value: "true"
+- **errors-as-warnings:** _(Optional)_ - Downgrade errors to warnings
+  - Expected value: "true"
+- **prepend-filename** _(Optional)_ - Prepend error 'filename' attribute to the message
+  - Expected value: "true"
+- **prepend-source** _(Optional)_ - Prepend error 'source' attribute to the message
+  - Expected value: "true"
```
Comment (staabm) [inline README.md:221]:
```suggestion
```

**Comment by staabm** [comment]
Thanks
